### ISSUE TITLE
dotnet cli failing after installing vs2017

### ISSUE THREAD

**Issue by dazinator** [dotnet/cli#6611]
## Steps to reproduce
I have a build agent that was building a VS2015, project json based solution. I have installed vs2017 on the buildagent along side vs2015, with the asp.net core workload. During our build of the VS2015 solution we call the "dotnet cli", and now some steps are failing. For example, `dotnet pack` complains that there is a missing targets and props file.

![image](https://cloud.githubusercontent.com/assets/3176632/26117923/fc6c4678-3a5f-11e7-8789-c5671bfa644f.png)

For some reason its looking for the targets and props files under the sdk directory, and the version of the dotnet cli being used is now 1.0.4 which doesn't seem to have those targets there. The targets exist instead within this folder: "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\DotNet".

## Expected  behaviour
Installing VS2017 alongside VS2015 shouldn't break existing vs2015 builds that were using the CLI previously. If this can't happen for technical reasons, the installer for the CLI should probably warn you that upgrading the version of the CLI may break existing builds. 
If I copy the targets into the expected location, or pass in a custom msbuild property `/p:VSToolsPath="C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0"` then `dotnet pack` fails with a different error:

```
C:\Users\darrell.tunnell\Source\Repos\gluon2\gluon\src\Gluon.Core\Gluon.Core.xproj : error MSB4057: The target "pack" does not exist in the project.
```

So basically, msbuild is happy now, but the cli expects the project file to be using the new vs2017 build targets, and so expects to be able to call into an msbuild "Pack" target which obviously doesn't exist. Could it not maintain backwards compatibility here?

Is there a recommended approach for having old style (project json) and new style (msbuild) builds happening on the same machine? 

## Actual behavior
Builds are broken when CLI is updated.

## Environment data
`dotnet --info` output: as per screenshot above.

**Comment by dazinator** [comment]
Also I have a question, what installer do I need to run, to install a version of the CLI that is capable of building the VS2015 / project.json / xproj based solution?

**Comment by omajid** [comment]
@dazinator Maybe remove all .NET Core installs? You can project.json/xproj support by getting any of the builds [here](https://github.com/dotnet/core/blob/master/release-notes/download-archive.md) with a pre-release SDK version (RC4, Preview 2, that sort of thing). [For example 1.0.3 + SDK RC4](https://github.com/dotnet/core/blob/master/release-notes/download-archives/rc4-download.md).

**Comment by livarcocc** [comment]
the CLI is side-by-side, which means that your preview2 installation is still available in your box. However, when finding an SDK to execute, our policy is to always rollforward to the highest available version. So, if you have 1.0.0-preview-003133 and 1.0.4. 1.0.4 is a highest version and will be picked up.

You can change that by adding a global.json file in the root of your repo, like this:

{
  "sdk": {
    "version": "1.0.0-preview2-003133"
  }
}

**Comment by dazinator** [comment]
Thank you. It turns out I had specified a version in global.json, but the matching version of the sdk wasn't installed on the build server, In this scenario rather than failing it was just defaulting to the latest SDK still.  

> However, when finding an SDK to execute, our policy is to always rollforward to the highest available version.

Makes sense I guess.. Would be better though if the policy was clever enough to only roll forward to the latest "compatible" version from the target project. For example if the project was build targeting with cli 1.0.0 cli compatibility, then executing `dotnet` cli commands against such a project wouldn't pick cli version `2.0.0`.  Perhaps then you could also use semver to signal breaking changes.
